R/rw.R

```diff
@@ -0,0 +1,229 @@
+#' Reweighting factor container
+#'
+#' This function `rw()` creates containers for reweighting factors
+#' of class `rw`. This class is particularly designed to store reweighting 
+#' factors for each gauge configuration, that can be applied on 
+#' correlation functions emerging in statistical and quantum field theory
+#' simulations. Arithmetic operations are defined for this class in
+#' several ways, as well as for increasing statistics and \link{is.rw}.
+#'
+#' @details
+#'#'
+#' @family rw constructors
+#'
+#' @export
+rw <- function () {
+  rw <- list()
+  class(rw) <- append(class(rw), 'rw')
+  return (rw)
+}
+
+#' rw metadata mixin constructor
+#'
+#' @param .rw `rw` object to extend.
+#' @param conf.index list of Integers, containing the index of gauge configurations
+#'
+#' @family cf constructors
+#'
+#' @export
+rw_meta <- function (.rw = rw(), conf.index = NA) {
```

Comment:
conf.index should not have a default argument, I would say, otherwise what's the point?

Reply:
is this fixed?

Reply:
yes